### DISCUSSION TITLE
Nested/Embedded datastructures possible?

### DISCUSSION THREAD

**Discussion by alwins0n** [com-lihaoyi/scalasql#13] [General]
Thank you for creating this!

Would it be possible to support nested or "embedded" case classes within a case class representation of a table?

non compiling example:

```scala
case class Customer[T[_]](
    name: T[String],
    address: T[Address[T[_]]]
)

object Customer extends Table[Customer]

case class Address[T[_]](
    street: T[String],
    city: T[String],
    zipCode: T[String],
    country: T[String]
)
```

the expected outcome in this case would be that the address value is treated as if it were flat in the customer table. this would allow sharing or reusing data structures in different contexts. 

a probably better example, would be `MonetaryAmount` with a value and currency component. its used a lot in different places if an app commits to storing values with currency.

Maybe there is some way to hook into something like a `CompositeTypeMapper` where one can supply custom givens analogous to the `TypeMapper` typeclass

for reference, I asked a somewhat similar question in another repo: https://github.com/AugustNagro/magnum/issues/24 obviously, it's tough for me to let go of that feature after years of using java's hibernate :)

**Comment by lihaoyi** [comment]
@alwins0n I think https://github.com/com-lihaoyi/scalasql/blob/main/docs/reference.md#datatypesenclosing is what you want

**Reply by alwins0n** [reply]
thank you. I don't know how I missed that.
